opm/parser/eclipse/Deck/DeckView.hpp

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright 2021 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef DECKVIEW_HPP
+#define DECKVIEW_HPP
+
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+
+namespace Opm {
+
+
+class DeckView {
+public:
+    typedef std::vector<std::reference_wrapper<const DeckKeyword>> storage_type;
+
+
+    struct Iterator : public storage_type::iterator {
+        Iterator(storage_type::const_iterator inner_iter) :
+            inner(inner_iter)
+        {}
+
+        const DeckKeyword& operator*()  { return this->inner->get(); }
+        const DeckKeyword* operator->() { return &this->inner->get(); }
+
+        Iterator& operator++()    { this->inner++; return *this; }
+        Iterator  operator++(int) { Iterator tmp = *this; ++(*this); return tmp; }
+
+        Iterator& operator--()    { this->inner--; return *this; }
```

Comment:
Predecrement should be implemented in terms of predecrement of the contained storage iterator.